stream_exercise.py

```diff
@@ -28,40 +26,32 @@ class StreamProcessor(object):
                together: if it reaches the 10th two digit number and the
                sum has not yet reached 200, then the method will stop and
                return 10.
+        """
 
-    For example, given a stream yielding "234761640930110349378289194", the
-    process method will:
-
-            1. Read two digits at a time from the stream: "23", "47", "61", etc.
-            2. Convert these digits into a number: 23, 47, 61, etc., and  make a
-               running total of these numbers: 23 + 47 equals 70. 70 + 61 equals
-               131, etc.
-            3. For this particular stream, the running total will exceed 200 after
-               5 such additions: the `process` method should return 5.
-
-    You can see the `tests.py` file for more examples of expected outcomes.
-    """
+        count = 0  # How many two-digit numbers the `process` method has added together.
+        total = 0  # The running total of sums.
 
-    def __init__(self, stream):
-        self._stream = stream
+        # I did this before looking at the solution.  Works the same, perhaps not as
+        #   clean as the other, but thought I'd leave it as I wrote it.
 
-    def process(self):
-        """
-        TODO: Implement the `process` method, as described above.
-        
-        :return: int
-        """
+        for block in iter(lambda: self._stream.read(2), ""):
 
-        count = 0  # How many two-digit numbers the `process` method has added
-                   # together.
-        total = 0  # The running total of sums.
+            # Because in his test file, if there's only a single digit left, it's ignored.
+            if len(block) < 2:
+                return count
+            count += 1
+            total += int(block)
+            if total >= 200 or count == 10:
+                return count
+        return count
 
-        # TODO: WRITE CODE HERE:
 
-        # Just some example syntax, you can read two digits from the head of the
-        # stream using the following code:
-        #
-        # digits = self._stream.read(2)
+def main():
+    """ We Wuz Main """
+    import io 
```

Comment:
imports should go at the top of the file unless your trying to avoid circular imports when doing recursion.

Reply:
In this case, i placed the import in main so that it would only be imported if it was run directly, and not when the file is imported from another file.

Reply:
great point.